File: pset1/mario/mario.c
#include <stdio.h>
#include <cs50.h>

void build(int number, string character);

int main(void)
{
    // get the pyramid height
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // brick char
    string brick = "#";
    // space char
    string space = " ";

    // build pyramid
    int count = 1;
    while (count <= height)
    {
        build(height - count, space);
        build(count, brick);
        build(2, space);
        build(count, brick);
        printf("\n");
        count++;
    }
}

void build(int number, string character)
{
    int count = 0;
    while (count < number)
    {
        printf("%s", character);
        count++;
    }
}

File: pset6/mario.py
from cs50 import get_int

# get pyramids height from user
while True:
    height = get_int("Height: ")
    # height must be integer between 1 and 8, inclusive.
    if height > 0 and height < 9:
        break

for i in range(1, height + 1):
    for j in range(1, height + 1):
        if j > (height - i):
            # print brick
            print("#", end="")
        else:
            # print the blank space before bricks
            print(" ", end="")
        if j == height:
            # print the gap
            print("  ", end="")
            # print the second part of pyramid
            print("#" * i, end="")
    print("")
